sokol_app.h

```diff
@@ -6458,6 +6462,35 @@ _SOKOL_PRIVATE LRESULT CALLBACK _sapp_win32_wndproc(HWND hWnd, UINT uMsg, WPARAM
                     }
                 }
                 break;
+
+
+            // Update window's DPI and size if its moved to another monitor with a different DPI
+            // Only send by windows, if PROCESS_PER_MONITOR_DPI_AWARE:
+            case WM_DPICHANGED:
+            {
+                HINSTANCE user32 = LoadLibraryA("user32.dll");
+
+                if (user32) {
+                    typedef UINT(WINAPI * GETDPIFORWINDOW_T)(HWND hwnd);
+                    GETDPIFORWINDOW_T fn_getdpiforwindow = (GETDPIFORWINDOW_T)(void*)GetProcAddress(user32, "GetDpiForWindow");
+
+                    if (fn_getdpiforwindow) {
+
+                        UINT dpix = fn_getdpiforwindow(_sapp.win32.hwnd);
+                        _sapp.win32.dpi.window_scale = (float)dpix / 96.0f;
+                        _sapp.win32.dpi.content_scale = _sapp.win32.dpi.window_scale;
+                        _sapp.dpi_scale = _sapp.win32.dpi.window_scale;
+
+                        LPRECT lpRect = (LPRECT)lParam;
+                        SetWindowPos(hWnd, nullptr, lpRect->left, lpRect->top, lpRect->right - lpRect->left, lpRect->bottom - lpRect->top, SWP_NOZORDER | SWP_NOACTIVATE);
+                    }
+
+                    FreeLibrary(user32);
+                }
```

Comment:
I think it would be better to wrap this code into a small helper function _sapp_win32_dpi_changed().